pkg/cloudprovider/vsphereparavirtual/controllers/routablepod/ippool/ippool_controller.go

```diff
@@ -215,10 +215,7 @@ func (c *Controller) processIPPoolCreateOrUpdate(ippool *ippoolv1alpha1.IPPool)
 	// make map of allocated subnets
 	subs := make(map[string]string)
 	for _, sub := range ippool.Status.Subnets {
-		// if ippool is exhausted, the cidr value could be empty
-		if sub.CIDR != "" {
-			subs[sub.Name] = sub.CIDR
-		}
+		subs[sub.Name] = sub.CIDR
```

Comment:
does this aim to avoid 'null'?
and we just display empty values in status (if CIDR is empty)?

Reply:
There are cases like cidr exhaustion, the value of CIDR in status will be empty string. I set this to make sure that status.cidr will be source of truth, regardless it's value.
If NCP fail to allocate cidr, then the podCIDR should have the same value.